File: Ex4/Task4.3/DegreeCategory.scala
/**
 * Takes a mark as a command line argument
 * and prints in which degree category it falls into
 */
object DegreeCategory {
    def main(args: Array[String]) {
        val degree: String = args(0).toDouble match {
            case x if (x < 32) => "Fail"
            case x if (x < 40) => "Pass / ordinary degree"
            case x if (x < 50) => "Honorous, third class"
            case x if (x < 60) => "Honorous, second class, division two"
            case x if (x < 70) => "Honorous, second class, division one"
            case _ => "Honorous, first class"
        }

        println(s"Degree category: $degree")
    }
}

File: ex4/task4.3/DegreeCategory.java
/**
* Prints out the degree earned by the student, uses if else statements.
* Tested for 32, 40, 51, 69, 70. Expected results given.
* @author Vlad Georgescu 
*/
public class DegreeCategory
{
    public static void main(String[] args)
    {
        double studentGrade = Double.parseDouble (args[0]);

        if (studentGrade < 32) {
            System.out.println("Fail!");
        } else if (studentGrade < 40) {
                System.out.println("Pass/ordinary degree!");
        } else if (studentGrade < 50) {
                System.out.println("Honours, third class!");
        } else if (studentGrade < 60) {
                System.out.println("Honours, second class, divion two!");
        } else if (studentGrade < 70) {
                System.out.println("Honours, second class, division one!");
        } else {
                System.out.println("Honours, firts class");
        } 
    }
}